src/H5Sselect.c

```diff
@@ -95,10 +90,14 @@ H5S_select_offset(H5S_t *space, const hssize_t *offset)
     /* Check args */
     assert(space);
     assert(0 < space->extent.rank && space->extent.rank <= H5S_MAX_RANK);
-    assert(offset);
 
-    /* Copy the offset over */
-    H5MM_memcpy(space->select.offset, offset, sizeof(hssize_t) * space->extent.rank);
+    /* Copy the offset over. As a special case, when offset is NULL, we
+     * reset all dimensions to zero.
+     */
+    if (offset)
+        H5MM_memcpy(space->select.offset, offset, sizeof(hssize_t) * space->extent.rank);
+    else
+        memset(space->select.offset, 0, sizeof(hssize_t) * space->extent.rank);
 
     /* Indicate that the offset was changed */
     space->select.offset_changed = true;
```

Comment:
should offset_changed be set to FALSE if NULL was passed?

Reply:
We don't do that when you pass in all zeroes, so that would be something slightly new under the hood, but it could be done. We'd probably want to do that in the all zeroes case, too.

Reply:
Wouldn't we want to set it to false only when the offset was already 0?

Reply:
I think the offset is only non-zero when you've moved the selection around.